### ISSUE TITLE
Since/Until

### ISSUE THREAD

**Issue by Egorand** [square/wire#1242]

**Comment by swankjesse** [comment]
```
message Customer {
  optional age = 1 [until = 20191130]; // Deprecated, use birth_year instead
  optional birth_year = 2 [since = 20190601];
}
```

**Comment by swankjesse** [comment]
https://github.com/square/wire/pull/1316

Next steps:

 * [ ] enums
 * [ ] rename IdentifierSet to PruningRules
 * [ ] decide whether to use longs or strings for versions
 * [ ] decide whether to scope versions to packages
 * [ ] decide what label to use for oldest..newest in Gradle (oldest_version newest_version?)

**Comment by oldergod** [comment]
Fixed with #1481
